components/ws-manager/pkg/manager/pod_controller.go

```diff
@@ -67,6 +68,21 @@ func (r *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 // SetupWithManager sets up the controller with the Manager.
 func (r *PodReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
+		WithEventFilter(
+			predicate.Funcs{
+				UpdateFunc: func(e event.UpdateEvent) bool {
+					_, ok := e.ObjectNew.GetAnnotations()[workspaceIDAnnotation]
+					return ok
+				},
+				CreateFunc: func(e event.CreateEvent) bool {
+					_, ok := e.Object.GetAnnotations()[workspaceIDAnnotation]
+					return ok
+				},
```

Comment:
We can remove [these lines](https://github.com/gitpod-io/gitpod/blob/7136c68be3b0409473d351a88dd68c41e1ae5be7/components/ws-manager/pkg/manager/pod_controller.go#L54-L57) now.

Reply:
https://github.com/gitpod-io/gitpod/pull/15475/files#r1058569770